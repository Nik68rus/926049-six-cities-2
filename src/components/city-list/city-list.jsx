import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer';

const CityList = (props) => {
  const {currentCity, cities, changeCityClickHandler} = props;
  return <ul className="locations__list tabs__list">
    {cities.map((city) => (
      <li className="locations__item" key={`city-${city}`}>
        <a
          className={`locations__item-link tabs__item ${city === currentCity && `tabs__item--active`}`}
          href="#"
          onClick={() => changeCityClickHandler(city)}>
          <span>{city}</span>
        </a>
      </li>
    )
    )}
  </ul>;
};

CityList.propTypes = {
  currentCity: PropTypes.string.isRequired,
  cities: PropTypes.array.isRequired,
  changeCityClickHandler: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentCity: state.city,
  offers: state.offers,
  cities: state.cityNames,
});

const mapDispatchToProps = (dispatch) => ({
  changeCityClickHandler: (city) => {
    dispatch(ActionCreator.changeCity(city));
    dispatch(ActionCreator.getOffers(city));
  }
});

export {CityList};

export default connect(mapStateToProps, mapDispatchToProps)(CityList);
