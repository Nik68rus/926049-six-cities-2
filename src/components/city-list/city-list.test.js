import renderer from 'react-test-renderer';
import {CityList} from '../city-list/city-list';

it(`City list correctly renders after relaunch`, () => {
  const cityList = renderer
    .create(<CityList
      currentCity={{
        name: ``,
        location: {
          latitude: 0,
          longtitude: 0,
          zoom: 0,
        }
      }}
      cities={[]}
      changeCityClickHandler={jest.fn()}
    />)
  .toJSON();

  expect(cityList).toMatchSnapshot();
});
