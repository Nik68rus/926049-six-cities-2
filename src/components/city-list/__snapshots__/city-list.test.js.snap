// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City list correctly renders after relaunch 1`] = `
<ul
  className="locations__list tabs__list"
/>
`;
